cql3/statements/select_statement.cc

```diff
@@ -421,7 +421,7 @@ select_statement::do_execute(query_processor& qp,
     const bool aggregate = _selection->is_aggregate() || has_group_by();
     const bool nonpaged_filtering = _restrictions_need_filtering && page_size <= 0;
     if (aggregate || nonpaged_filtering) {
-        page_size = page_size <= 0 ? internal_paging_size : std::min(page_size, internal_paging_size);
+        page_size = page_size <= 0 ? internal_paging_size : page_size;
```

Comment:
This applies to the the internal pages read while aggregating, yes? Not the result. If so there's no problem to drop it and it may improve performance a little.

A bigger task is to page to the user. We could consider sending the accumulators back to the user, and so be able to pause a query mid-aggregation. That prevents timing out when aggregating a large partition. Though it's a poor complexity/benefit tradeoff .

Reply:
Yes, this applies to the internal pages read while aggregating, not the result. What do you mean by "drop it"?

What you call "a bigger task" is not addressed by this PR at all. For the lack of a better CQL API, aggregating SELECT could return empty pages with the partial accumulation in the paging state (so not visible to the user). But this is a separate idea, unrelated to what this PR does. By the way, in DynamoDB the API addresses this issue more directly: The aggregation (the only one supported is COUNT) returns the aggregation for a single page of results (i.e., the count of items in this page) - the user controls this page size - and the user is responsible for adding up all these partial sums.

Reply:
drop the std::min() call.